src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.BIO.Custom.cs

```diff
@@ -0,0 +1,163 @@
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+using System.Diagnostics;
+
+internal static partial class Interop
+{
+    internal static class CustomBio
+    {
+        private static IntPtr _customBioMethodStructure;
+        private static CreateDelegate _createDelegate;
+        private static DestroyDelegate _destroyDelegate;
+        private unsafe static ControlDelegate _controlDelegate;
+        private unsafe static ReadDelegate _readDelegate;
+        private unsafe static WriteDelegate _writeDelegate;
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private unsafe delegate int CreateDelegate(bio_st* bio);
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private unsafe delegate int ReadDelegate(IntPtr bio, void* buf, int size);
```

Comment:
Struct wrapper for these types to make it strongly typed? e.g.
```csharp
struct Bio
{
    IntPtr bioPtr;
}
```
The wrapper should happily go through the function pointer